Packet++/src/WireGuardLayer.cpp

```diff
@@ -0,0 +1,324 @@
+#define LOG_MODULE PacketLogModuleWireGuardLayer
+
+#include "UdpLayer.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "WireGuardLayer.h"
+#include "Logger.h"
+#include <sstream>
+#include <iomanip>
+#include <cstring>
+
+namespace pcpp
+{
+	void WireGuardLayer::parseNextLayer()
+	{
+		size_t headerLen = getHeaderLen();
+		if (m_DataLen <= headerLen || headerLen == 0)
+			return;
+		m_NextLayer = WireGuardLayer::parseWireGuardLayer(m_Data, m_DataLen, this, m_Packet);
```

Comment:
`parseNextLayer` should always generate a layer. If `m_DataLen <= headerLen` (and it's not 0) or if `WireGuardLayer::parseWireGuardLayer()` returns `nullptr`, this method should generate a `PayloadLayer`

Reply:
Sure, I've made the changes.